diesel/src/sqlite/types/json.rs

```diff
@@ -0,0 +1,164 @@
+extern crate serde_json;
+
+use crate::deserialize::{self, FromSql};
+use crate::serialize::{self, IsNull, Output, ToSql};
+use crate::sql_types;
+use crate::sqlite::{Sqlite, SqliteValue};
+
+#[cfg(all(feature = "sqlite", feature = "serde_json"))]
+impl FromSql<sql_types::Json, Sqlite> for serde_json::Value {
+    fn from_sql(value: SqliteValue<'_, '_, '_>) -> deserialize::Result<Self> {
+        serde_json::from_str(value.read_text()).map_err(|_| "Invalid Json".into())
+    }
+}
+
+#[cfg(all(feature = "sqlite", feature = "serde_json"))]
+impl ToSql<sql_types::Json, Sqlite> for serde_json::Value {
+    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Sqlite>) -> serialize::Result {
+        out.set_value(serde_json::to_string(self)?);
+        Ok(IsNull::No)
+    }
+}
+
+#[cfg(all(feature = "sqlite", feature = "serde_json"))]
+impl FromSql<sql_types::Jsonb, Sqlite> for serde_json::Value {
+    fn from_sql(value: SqliteValue<'_, '_, '_>) -> deserialize::Result<Self> {
+        let bytes = value.read_blob();
+        if bytes[0] != 1 {
+            return Err("Unsupported JSONB encoding version".into());
+        }
+        serde_json::from_slice(&bytes[1..]).map_err(|_| "Invalid Json".into())
```

Comment:
@weiznich I might have misunderstood something, but how does Pg's support for JSONB work then?

https://github.com/diesel-rs/diesel/blob/f7e9819303877b5a255143c4b32fa1e5745aaad0/diesel/src/pg/types/json.rs#L35

As far as I can tell, it reads the first byte then assumes the rest is a valid JSON UTF-8 String. Or am I missing something?

Reply:
I have updated my PR with commits to implement a proper SQLite JSONB reader/writer. Including unit tests.

Reply:
The test does not work because there are missing parts in the implementation:

* We need to include this module for sqlite as well: https://github.com/diesel-rs/diesel/blob/master/diesel/src/type_impls/mod.rs#L5
* We need to allow mapping to jsonb for sqlite as well: https://github.com/diesel-rs/diesel/blob/master/diesel/src/type_impls/json.rs#L12